protollm/connectors/rest_server.py

```diff
@@ -0,0 +1,113 @@
+import json
+from typing import Any, Dict, List, Optional, Union
+
+import requests
+from langchain_core.callbacks import CallbackManagerForLLMRun
+from langchain_core.language_models import BaseChatModel
+from langchain_core.messages import (AIMessage, BaseMessage, HumanMessage,
+                                     SystemMessage)
+from langchain_core.outputs import ChatGeneration, ChatResult
+
+
+class ChatRESTServer(BaseChatModel):
+    model: Optional[str] = 'llama3'
+    base_url: str = 'http://10.32.2.2:8672'
+    timeout: Optional[int] = None
+    """Timeout for the request stream"""
```

Comment:
Google-style docstring would be more informative

Reply:
This style is used in langchain